sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuTransitionOverrides.scala

```diff
@@ -269,7 +269,11 @@ class GpuTransitionOverrides extends Rule[SparkPlan] {
       if (conf.exportColumnarRdd) {
         updatedPlan = detectAndTagFinalColumnarOutput(updatedPlan)
       }
-      if (conf.isTestEnabled) {
+      val adaptiveQueryEnabled = plan.conf
+          .getConfString("spark.sql.adaptive.enabled", "false").toBoolean
+      if (conf.isTestEnabled && !adaptiveQueryEnabled) {
+        // we cannot assert that the plan is on the GPU at this stage when adaptive query
```

Comment:
might be good to put one or two examples here on what nodes are no longer on the GPU when adaptive execution is turned on? Or perhaps point to a github issue.

Reply:
I changed the approach here. There are now checks that specifically ignore the operators that AQE adds (AdaptiveSparkPlanExec, QueryStageExec, CustomShuffleReaderExec)

Reply:
:+1: